### PR TITLE
fix: read and apply per target Cxx settings and consolidate private includes

### LINKED ISSUE #674: GEOSwift (geos) package fails to compile when using rules_swift_package_manager but not Xcode
I'm trying to use the GEOSwift package (https://github.com/GEOSwift/GEOSwift) which depends on the geos package (https://github.com/GEOSwift/geos) via rules_swift_package_manager and getting a bunch of compiler errors due to some c++ stdlib functions it uses:
```
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/cmath:363:1: note: using declaration annotated with 'using_if_exists' here
using ::ceil _LIBCPP_USING_IF_EXISTS;
^
In file included from external/rules_swift_package_manager~0.15.0~swift_deps~swiftpkg_geos/Sources/geos/src/algorithm/Angle.cpp:21:
In file included from external/rules_swift_package_manager~0.15.0~swift_deps~swiftpkg_geos/Sources/geos/include/geos/algorithm/Angle.h:22:
In file included from external/rules_swift_package_manager~0.15.0~swift_deps~swiftpkg_geos/Sources/geos/include/geos/algorithm/Orientation.h:22:
In file included from external/rules_swift_package_manager~0.15.0~swift_deps~swiftpkg_geos/Sources/geos/include/geos/geom/Coordinate.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/set:1601:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/functional:515:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/boyer_moore_searcher.h:25:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/unordered_map:523:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__hash_table:2241:83: error: reference to unresolved using declaration
                  std::__is_hash_power2(__bc) ? std::__next_hash_pow2(size_t(std::ceil(float(size()) / max_load_factor()))) :
```
I've tried setting each of the following in my .bazelrc and got different errors each time:
```
build --cxxopt='-std=c++11'
build --cxxopt='-std=c++14'
build --cxxopt='-std=c++20'
```
This package builds and runs fine when I use it through Xcode 15 as a swift package dependency, so the issue is on the Bazel side.

Per our discussion on Slack, this is because rules_swift_package_manager is not processing the target-specific cxxSettings from the Package.swift files.

### PR DESCRIPTION
**Pull request by cgrindel**
- Read and apply Cxx settings for Swift package targets.
- Consolidate the list of private includes for clang targets to smallest possible number by removing those with a common parent leaving the shortest of the common parent paths.
- Add [GEOSwift package](https://github.com/GEOSwift/GEOSwift) to the `interesting_deps` example.

Closes #674.

### CODE REVIEW

**Comment by cgrindel** [comment]
cc: @AttilaTheFun

**Comment by AttilaTheFun** [comment]
Thanks @cgrindel ! I verified that the GEOSwift targets build with:
```
git_override(
    module_name = "rules_swift_package_manager",
    commit = "b6b73e44e094b5c4d6cfc2f54d9aa9dc90182e93",
    remote = "https://github.com/cgrindel/rules_swift_package_manager",
)
```

**Review by luispadron — APPROVED** [review]

File: swiftpkg/internal/pkginfos.bzl [hunk 1]
```diff
@@ -910,11 +929,16 @@ def _new_clang_src_info_from_sources(
...
-        extra_hdr_dirs.extend(lists.flatten([
-            [paths.join(target_path, path) for path in bs.values]
-            for bs in clang_settings.hdr_srch_paths
-        ]))
+
+    # if clang_settings != None:
+    #     extra_hdr_dirs.extend(lists.flatten([
+    #         [paths.join(target_path, path) for path in bs.values]
+    #         for bs in clang_settings.hdr_srch_paths
+    #     ]))
```
Comment (luispadron) [inline swiftpkg/internal/pkginfos.bzl:937]:
```suggestion
```

File: swiftpkg/internal/pkginfos.bzl [hunk 2]
```diff
@@ -279,6 +282,13 @@ def _new_target_from_json_maps(
             ]
 
     elif module_type == module_types.clang:
+        # other_hdr_srch_paths = clang_settings.hdr_srch_paths + \
+        #                        cxx_settings.hdr_srch_paths
```
Comment (luispadron) [inline swiftpkg/internal/pkginfos.bzl:286]:
```suggestion
```

**Pull request merged**